test_cases/README.md

```diff
@@ -1,31 +1,35 @@
 ## Regression tests for typeshed
 
-This directory contains regression tests for the stubs found elsewhere in the
-typeshed repo.
+This directory contains code samples that act as a regression test for the
+standard library stubs found elsewhere in the typeshed repo.
 
-**This directory should *only* contain tests for functions and classes which
+**This directory should *only* contain test cases for functions and classes which
 are known to have caused problems in the past, where the stubs are difficult to
```

Comment:
Is there, or should there be, a way to map the tests to when they've caused issues in the past (e.g. the GitHub issue or merge request(s))? Not something that would make sense to retroactively hunt down, but may be useful when including new test cases.

Reply:
The reason I ask was not due to quantity of relationships, but accessibility, primarily. I don't have a problem with this if everyone is happy with it, but I think it's worth considering if being more deliberate about using this information to group test cases and provide feedback in test failures about what behavior regressed could be a useful longer term goal.

Reply:
That's an interesting point, for sure. There are things we could consider there (like maybe using an actual test runner, such as pytest) -- but I do also value simplicity of implementation, and readability. So there's trade-offs here: making the output from the test more readable might make the implementation of the test more complex and less readable.

One thing we could consider: `assert_type` is a very new function. Perhaps we could make it more user-friendly: maybe there could be an optional `msg` argument that the type-checker could report if the "static assertion" failed.